### DISCUSSION TITLE
How does testing on Physical device before moving to production and app store release work ?

### DISCUSSION THREAD

**Discussion by netkeojas** [serverpod/serverpod#1197] [Q&A] [answered]
Hi, I understand that to test on a physical device too we will need to have a server running either cloud or local server, but what I don't understand or rather not sure of is that is the flutter file only enough to run on the physical device or do we need the server file or client file as well ?
Few questions for production deployment / app store release.
1. The apk that we generate would be of only flutter file ?
2. How does the flutter communicate with the client/server side code in this case ?
3. Any suggestions on how to do testing on physical device before going to production (a complete setup or walkthrough apreciated).

**Comment by yahu1031** [comment, marked as answer]
@netkeojas At first, the server is isolated from the flutter app. And Client is a package that actually you use like any other package dependencies.

1. Yes, For your understanding, the client code too. Where it helps in serializing, communicating with endpoints/WS, etc...
2. It communicates with the help of the host you use in the `main.dart` (`http://localhost:8080`) or something like this. It will be the hosting domain when it comes to production testing (`https://api.examplepod.com`).
3. You can go through this for communicating through physical device #979

**Reply by netkeojas** [reply]
Thanks @yahu1031 , I understood some parts of it, I will go through the link that your have shared as well.
So as much I understood is this - 
We will only have flutter file inside the APK, and the place in main.dart is where the call will go to client of which the server is hosted at.
